.github/workflows/ci.yml

```diff
@@ -0,0 +1,61 @@
+name: ci
+on:
+  push:
+    branches:
+      - master
+  pull_request: {}
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macOS-latest, windows-latest]
+        ghc: ['8.6', '8.8', '8.10']
+        exclude:
+        - os: windows-latest
+          ghc: "8.8"
+        - os: windows-latest
+          ghc: "8.10"
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-haskell@v1.1.3
+      id: setup-haskell-cabal
+      with:
+        ghc-version: ${{ matrix.ghc }}
+        cabal-version: '3.2'
+    - name: Update cabal package database
+      run: |
+        cabal update
+        cabal freeze
+    - uses: actions/cache@v2
+      name: Cache cabal stuff
+      with:
+        path: |
+          ${{ steps.setup-haskell-cabal.outputs.cabal-store }}
+          dist-newstyle
+        key: ${{ runner.os }}-${{ matrix.ghc }}-${{ hashFiles('cabal.project.freeze') }}
+        restore-keys: |
+          ${{ runner.os }}-${{ matrix.ghc }}-
+    - name: Build
+      run: |
+        cabal build all
+    - name: Test
+      run: |
+        cd tests && cabal test
+    - name: Haddock
+      run: |
+        cabal haddock
+    - name: SDist
+      run: |
+        cabal sdist all
+    - name: Bench
+      run: |
+        cd bench
+        cabal bench bench-bytestring-builder -O0 --benchmark-options "--quick --min-duration=0 --include-first-iter"
+        cabal bench bench-builder-boundscheck -O0 --benchmark-options "--quick --min-duration=0 --include-first-iter"
+        cabal bench bench-builder-csv -O0 --benchmark-options "--quick --min-duration=0 --include-first-iter"
+        cabal bench bench-strict-indices -O0 --benchmark-options "--quick --min-duration=0 --include-first-iter"
```

Comment:
Why invoke each benchmark suite separately instead of having a single `cabal bench` command? With this setup we're more likely to forget testing any new benchmark suites we might add.

Reply:
Changed to a single `cabal bench`.